### ISSUE TITLE
Columns not available for extended Permissions model

### ISSUE THREAD

**Issue by isecchin** [spatie/laravel-permission#1836]
**Describe the bug**
Hi, first of all, I'm sorry if this is not a bug, but I'm having a hard time trying to get my app working with the latest release, so I'm not sure if it's a specific problem from my use case or if it's general.
Basically, recent releases kind of "broke" backwards compatibility with some features that are still documented as available, such as the ability to extend the models.
On my application I've extended both models (Permissions and Roles), mainly to have an extra attribute for the permissions (my app is divided between modules, so I have a `module_id` column on my permissions table that I double check, to make sure users can't use permissions of modules they don't have, build custom menus and so on).
I haven't updated the package for a few months now, and when I tried to update today, I encountered this issue, apparently in order to shrink the cache, the package has now the attributes of the permissions model hard-coded on the PermissionRegistrar, so I can't check for any other columns I have on the DB because they will never be available.
I'm not sure if this was intended, but if so, I guess we should update the docs and possibly document it as a breaking change (?).

**Versions**
- spatie/laravel-permission package version: 5.1.1
- illuminate/framework package version: 8.58.0

PHP version:
8.0.10

Database version:
MySQL 8.0.23

**To Reproduce**
Just add a new column to the permissions table and try to access it anywhere on the app using a method such as the `findByName` from the Permission's model.

**Expected behavior**
I guess it would be nice if we could have the attributes we want to cache on the config or something, so we can still shrink the cache in a way, but also have the flexibility to have extra columns, if needed.

**Comment by danilopinotti** [comment]
Hello!

I think It is caused by #1799 or #1568.
These PRs made changes in some things about serialization and cache.

**Comment by erikn69** [comment]
Just try `->findByName('name')->fresh()`

**Comment by isecchin** [comment]
Thank you for your feedback!

Yes, I've checked the previous issues and PRs and figured that this change was to resolve the cache size, I'm also using Vapor as described in one of the issues, but since I'm using Redis, I haven't ran into this issue yet, but I totally agree with the suggestion to shrink it.

@erikn69 correct me if I'm wrong, but your previous suggestion, although it would work, it would also make extra queries every time I needed to check for the `module_id`, skipping the cache entirely.

But it seems that your PR fix this, I just tried it out and I can now access extra attributes again, thank you!

**Comment by erikn69** [comment]
>it would also make extra queries every time I needed to check for the module_id, skipping the cache entirely

Yes, for that reason I better did a PR, I'm glad it works for you 👍

**Comment by PaolaRuby** [comment]
>Seems fixed!

@freekmurze  not fixed, you close the PR #1837 wich fix this

**Comment by erikn69** [comment]
@PaolaRuby i could try to fix this again, but probably will be discarded again
